frontend/apps/reader/modules/readerlink.lua

```diff
@@ -663,18 +679,36 @@ function ReaderLink:onTap(_, ges)
 end
 
 --- Remember current location so we can go back to it
-function ReaderLink:addCurrentLocationToStack()
-    if self.ui.document.info.has_pages then
-        table.insert(self.location_stack, self.ui.paging:getBookLocation())
-    else
-        table.insert(self.location_stack, {
-            xpointer = self.ui.rolling:getBookLocation(),
-        })
+--- increments location_stack_index to point the the last item,
+--- and clears the next item in case there was a Forward history
+function ReaderLink:addCurrentLocationToStack(loc)
+    local location = loc
+    if location == nil then
+        if self.ui.document.info.has_pages then
+            location = self.ui.paging:getBookLocation()
+        else
+            location = {xpointer = self.ui.rolling:getBookLocation(),}
+        end
     end
+    self.location_stack_index = self.location_stack_index + 1
+    self.location_stack[self.location_stack_index] = location
+    self.location_stack[self.location_stack_index+1] = nil
```

Comment:
Are we supposed somehow somewhere to navigate again to older forward locations, when you got back and follow a new path ?
It's a bit unclear this gymnastic with `+1 = nil` or `~= nil`.